src/frontend/src/hooks/finance.hooks.ts

```diff
@@ -3,23 +3,99 @@
  * See the LICENSE file in the repository root folder for details.
  */
 import { useMutation, useQuery } from 'react-query';
-import { getSingleReimbursementRequest, uploadSingleReceipt } from '../apis/finance.api';
-import { ReimbursementRequest } from 'shared';
+import {
+  createReimbursementRequest,
+  getAllExpenseTypes,
+  getAllVendors,
+  uploadSingleReceipt,
+  getSingleReimbursementRequest
+} from '../apis/finance.api';
+import { ClubAccount, ExpenseType, ReimbursementProductCreateArgs, ReimbursementRequest, Vendor } from 'shared';
+
+export interface CreateReimbursementRequestArgs {
+  vendorId: string;
+  account: ClubAccount;
+  dateOfExpense: Date;
+  expenseTypeId: string;
+  reimbursementProducts: ReimbursementProductCreateArgs[];
+  receiptFiles: {
+    file: File;
+  }[];
+  totalCost: number;
+}
 
 /**
  * Custom React Hook to upload a new picture.
+ */
+export const useUploadSingleReceipt = () => {
+  return useMutation<{ googleFileId: string; name: string }, Error, { file: File; id: string }>(
+    ['finance', 'image'],
+    async (formData: { file: File; id: string }) => {
+      const { data } = await uploadSingleReceipt(formData.file, formData.id);
+      return data;
+    }
+  );
+};
+
+/**
+ * Uploads many receipts to a given reimbursement request
+ *
+ * @returns The created receipt information
+ */
+export const useUploadManyReceipts = () => {
+  return useMutation<{ googleFileId: string; name: string }[], Error, { files: File[]; id: string }>(
+    ['finance', 'image'],
```

Comment:
I dont have access to this yet i only get the form data inside the mutate async

Reply:
?, you have id right there so it'd be ['reimbursement-requests', id]

Reply:
negative that doesnt work

Reply:
💀 , it's bc we're passing it in through the payload going through the useMutation instead of through a parameter to the hook. Pls don't resolve if we haven't actually solved the problem